packages/xdl/src/project/ProjectUtils.js

```diff
@@ -126,203 +118,64 @@ export function attachLoggerStream(projectRoot: string, stream: any) {
   _getLogger(projectRoot).addStream(stream);
 }
 
-export async function fileExistsAsync(file: string): Promise<boolean> {
-  try {
-    return (await fs.stat(file)).isFile();
-  } catch (e) {
-    return false;
-  }
-}
-
-export function resolveModule(request, projectRoot, exp) {
-  const fromDir = exp.nodeModulesPath ? exp.nodeModulesPath : projectRoot;
-  return resolveFrom(fromDir, request);
-}
-
-async function _findConfigPathAsync(projectRoot) {
-  const appJson = path.join(projectRoot, 'app.json');
-  const expJson = path.join(projectRoot, 'exp.json');
-  if (await fileExistsAsync(appJson)) {
-    return appJson;
-  } else if (await fileExistsAsync(expJson)) {
-    return expJson;
-  } else {
-    return appJson;
-  }
-}
-
-let hasWarnedAboutExpJson = false;
+// Wrap with logger
 
 export async function findConfigFileAsync(
   projectRoot: string
 ): Promise<{ configPath: string, configName: string, configNamespace: ?string }> {
-  let configPath;
-  if (customConfigPaths[projectRoot]) {
-    configPath = customConfigPaths[projectRoot];
-  } else {
-    configPath = await _findConfigPathAsync(projectRoot);
-  }
-  const configName = path.basename(configPath);
-  const configNamespace = configName !== 'exp.json' ? 'expo' : null;
-
-  if (configName === 'exp.json' && !hasWarnedAboutExpJson) {
-    hasWarnedAboutExpJson = true;
-    logWarning(
-      projectRoot,
-      'expo',
-      `Warning: configuration using exp.json is deprecated.
-Please move your configuration from exp.json to app.json.
-Example app.json:
-{
-  "expo": {
-    (JSON contents from exp.json)
-  }
-}`
-    );
+  try {
+    return await ConfigUtils.findConfigFileAsync(projectRoot);
+  } catch (error) {
+    if (error instanceof ConfigUtils.ConfigError) {
+      logWarning(projectRoot, 'expo', error.message);
```

Comment:
Not sure if this error handling works as intended: when you catch the error and log a warning, this function returns `undefined` and then the code crashes soon after with `Cannot read property 'platforms' of null`. (When running `expo start` in a project with `exp.json` and no `app.json`)

However, `app.json` has been the default for more than two years now and now we've also had this deprecation warning live for almost two months, so I'd say let's drop the `catch` block here and let it be a hard error if you're still using `exp.json`.